atom/browser/web_contents_preferences.cc

```diff
@@ -89,23 +144,38 @@ WebContentsPreferences::~WebContentsPreferences() {
 bool WebContentsPreferences::SetDefaultBoolIfUndefined(
     const base::StringPiece& key,
     bool val) {
-  bool existing;
-  if (!dict_.GetBoolean(key, &existing)) {
-    dict_.SetBoolean(key, val);
+  auto* current_value =
+      preference_.FindKeyOfType(key, base::Value::Type::BOOLEAN);
+  if (current_value) {
+    return current_value->GetBool();
+  } else {
+    preference_.SetKey(key, base::Value(val));
     return val;
   }
-  return existing;
 }
 
 bool WebContentsPreferences::IsEnabled(const base::StringPiece& name,
                                        bool default_value) {
-  bool bool_value = default_value;
-  dict_.GetBoolean(name, &bool_value);
-  return bool_value;
+  auto* current_value =
+      preference_.FindKeyOfType(name, base::Value::Type::BOOLEAN);
+  if (current_value)
+    return current_value->GetBool();
+  return default_value;
 }
 
 void WebContentsPreferences::Merge(const base::DictionaryValue& extend) {
-  dict_.MergeDictionary(&extend);
+  if (preference_.is_dict())
```

Comment:
when would `preference_` not be a dict? it seems like `FindKeyOfType` will `CHECK()` if it isn't a dict.